typescript.js

```diff
@@ -2,79 +2,60 @@ module.exports = {
-        'no-dupe-class-members': 'off',
-        '@typescript-eslint/no-dupe-class-members': 'error',
-        'no-empty-function': 'off',
-        '@typescript-eslint/no-empty-function': ['error', {
-            allow: [
-                'arrowFunctions',
-                'functions',
-                'methods'
-            ]
-        }],
-        'no-extra-semi': 'off',
-        '@typescript-eslint/no-extra-semi': 'error',
-        'no-loop-func': 'off',
-        '@typescript-eslint/no-loop-func': 'error',
-        'no-redeclare': 'off',
-        '@typescript-eslint/no-redeclare': 'error',
-        'no-shadow': 'off',
-        '@typescript-eslint/no-shadow': 'error',
-        'no-unused-expressions': 'off',
-        '@typescript-eslint/no-unused-expressions': 'error',
-        'no-unused-vars': 'off',
-        '@typescript-eslint/no-unused-vars': ['error', { vars: 'all', args: 'after-used', ignoreRestSiblings: true }],
-        'no-use-before-define': 'off',
-        '@typescript-eslint/no-use-before-define': ['error', { functions: false, classes: true, variables: true }],
-        'no-useless-constructor': 'off',
-        '@typescript-eslint/no-useless-constructor': 'error',
-        quotes: 'off',
-        '@typescript-eslint/quotes': ['error', 'single', { avoidEscape: true }],
-        'no-return-await': 'off',
-        '@typescript-eslint/return-await': 'error',
-        semi: 'off',
-        '@typescript-eslint/semi': 'error',
-        'space-before-function-paren': 'off',
-        '@typescript-eslint/space-before-function-paren': ['error', {
-            anonymous: 'always',
-            named: 'never',
-            asyncArrow: 'always'
-        }],
-        'space-infix-ops': 'off',
-        '@typescript-eslint/space-infix-ops': 'error',
+
+        // None
+
+        /*
+            TypeScript rules outside of the recommended configuration
+        */
+        '@typescript-eslint/array-type': 'error',
+        '@typescript-eslint/ban-tslint-comment': 'error',
+        '@typescript-eslint/class-literal-property-style': 'error',
+        '@typescript-eslint/consistent-indexed-object-style': 'error',
+        '@typescript-eslint/consistent-type-assertions': 'error',
+        '@typescript-eslint/consistent-type-definitions': 'error',
+        '@typescript-eslint/consistent-type-imports': 'error',
+        '@typescript-eslint/explicit-function-return-type': 'error',
+        '@typescript-eslint/explicit-member-accessibility': 'error',
+        '@typescript-eslint/member-delimiter-style': 'error',
+        '@typescript-eslint/member-ordering': 'error',
+        '@typescript-eslint/method-signature-style': 'error',
+        '@typescript-eslint/no-confusing-non-null-assertion': 'error',
+        '@typescript-eslint/no-dynamic-delete': 'error',
+        '@typescript-eslint/no-extraneous-class': 'error',
+        '@typescript-eslint/no-implicit-any-catch': 'error',
+        '@typescript-eslint/no-invalid-void-type': 'error',
```

Comment:
[@typescript-eslint/no-invalid-void-type](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-invalid-void-type.md)

I'm still wading my way through the Document Manger lint errors, but one of the interesting examples that came up was this piece of an observable chain:
```
switchMap((webapp: WebApp): Observable<[Document, void]> => combineLatest([this.getDocumentFromWebApp(webapp), this.loadDocuments(this.activePlugin)])),
```

On a glance, it seems reasonable - it's a `combineLatest` where one of the observables is void. However, I'm not sure if there's a better method in RxJS for accomplishing the same thing while avoiding the void type. We could consider turning on `allowInGenericTypeArguments` for these use cases.

Reply:
@LukasKall said:
I'm not sure, but is this something where never can be used? Or does this cause an error because loadDocuments returns void?

Reply:
@TrevorKarjanis said:
Never implies that a value is never provided, like if the observable always errored in which case no values would ever be provided. I'll have to run this rule over my code, but assigning void to variables doesn't make much sense. In @mure 's case, combineLatest expects values, so unless it is mapped to a single value then undefined is probably correct.

Reply:
I don't think using undefined is possible in this case, since it has to be a type argument. I get a compiler error:
`Type 'void' is not assignable to type 'undefined'`

Reply:
@mure will continue researching this and come up with a proposal.

Reply:
I'm unable to get the `allowInGenericTypeArguments` config to work correctly (as far as I understand it). I've filed an issue here: https://github.com/typescript-eslint/typescript-eslint/issues/3143

Reply:
Based on the issue discussion it sounds like the behavior described is not a false positive so we may need a source change to support this rule. I'll mark this as resolved.